### PR TITLE
Add basic test for `dvc usages`

### PR DESCRIPTION
**Pull request by laurawarr**
Added a test and found some edge cases

### CODE REVIEW

File: src/utils/parsers/common.ts [hunk 1]
```diff
@@ -145,12 +145,21 @@ export abstract class BaseParser {
...
-            index += line.content.length - 1
-            if (index >= match.range.start && index <= match.range.end) {
+            const trimmedContent = line.content.trim()
+            if (!trimmedContent.length) continue
+
+            const { start: matchStartIndex, end: matchEndIndex } = match.range
+            const lineStartIndex = index
+            const lineEndIndex = lineStartIndex + trimmedContent.length - 1
+            if (
+                (matchStartIndex >= lineStartIndex && matchStartIndex <= lineEndIndex)
```
Comment (laurawarr) [inline src/utils/parsers/common.ts:157]: this handles the case that the match is contained entirely within the line

**Review by jsalaber — APPROVED** [review]
